cli/deployments/eirini.go

```diff
@@ -0,0 +1,285 @@
+	releaseDir, err := ioutil.TempDir("", "carrier")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(releaseDir)
+
+	releaseFile, err := helpers.UnEmbedFile(eiriniReleasePath)
+	if err != nil {
+		return errors.New("Failed to extract embedded file: " + eiriniReleasePath + " - " + err.Error())
+	}
+	defer os.Remove(releaseFile)
+
+	err = helpers.Untar(releaseFile, releaseDir)
+	if err != nil {
+		return err
+	}
+
+	message := "Creating Eirini namespace for core components"
+	out, err := helpers.SpinnerWaitCommand(message,
+		func() (string, error) {
+			file := path.Join(releaseDir, "deploy", "core", "namespace.yml")
+			return helpers.Kubectl("apply -f " + file)
+		},
+	)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("%s failed:\n%s", message, out))
+	}
+
+	message = "Creating Eirini namespace for workloads"
+	out, err = helpers.SpinnerWaitCommand(message,
+		func() (string, error) {
+			file := path.Join(releaseDir, "deploy", "workloads", "namespace.yml")
+			return helpers.Kubectl("apply -f " + file)
+
+		},
+	)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("%s failed:\n%s", message, out))
+	}
+
+	if err = k.patchNamespaceForQuarks(c, "eirini-workloads"); err != nil {
+		return err
+	}
+	if err = k.patchNamespaceForQuarks(c, "eirini-core"); err != nil {
+		return err
+	}
+
+	message = "Creating eirini certs using quarks-secret"
+	out, err = helpers.SpinnerWaitCommand(message,
+		func() (string, error) {
+			return helpers.KubectlApplyEmbeddedYaml(eiriniQuarksYaml)
+		},
+	)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("%s failed:\n%s", message, out))
+	}
+
+	for _, component := range []string{"core", "events", "metrics", "workloads"} {
+		message := "Deploying eirini " + component
+		out, err := helpers.SpinnerWaitCommand(message,
+			func() (string, error) {
+				dir := path.Join(releaseDir, "deploy", component)
+				return helpers.Kubectl("apply -f " + dir)
+			},
+		)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("%s failed:\n%s", message, out))
+		}
+	}
+
+	domain, err := options.GetString("system_domain", eiriniDeploymentID)
+	if err != nil {
+		return err
+	}
+	// Create secrets and services accounts
+	if err = k.createClusterRegistryCredsSecret(c, false); err != nil {
+		return err
+	}
+	if err = k.createClusterRegistryCredsSecret(c, true); err != nil {
+		return err
+	}
+	if err = k.createGitCredsSecret(c, domain); err != nil {
+		return err
+	}
+	if err = k.patchServiceAccountWithSecretAccess(c, "eirini"); err != nil {
+		return err
+	}
+
+	if err := c.WaitForPodBySelectorRunning("eirini-core", "name=eirini-api", 300); err != nil {
+		return errors.Wrap(err, "failed waiting Eirini api deployment to come up")
+	}
+	if err := c.WaitForPodBySelectorRunning("eirini-core", "name=eirini-metrics", 300); err != nil {
+		return errors.Wrap(err, "failed waiting Eirini metrics deployment to come up")
+	}
+
+	emoji.Println(":heavy_check_mark: Eirini deployed")
+
```

Comment:
Nit: Empty line

Reply:
A habit I got from [this linter](https://github.com/ssgreg/nlreturn). It's also a convention in other languages to make the return statement more visible. I would like to keep it, hoping in the future to add some linters to our project (a list: https://golangci-lint.run/usage/linters/). Is that ok?

Reply:
Sure. As I said, nits, mostly style. I am able to cope.